zkevm-circuits/src/evm_circuit/param.rs

```diff
@@ -3,7 +3,7 @@ use super::table::Table;
 // Step dimension
 pub(crate) const STEP_WIDTH: usize = 128;
 /// Step height
-pub const MAX_STEP_HEIGHT: usize = 16;
+pub const MAX_STEP_HEIGHT: usize = 21;
```

Comment:
I've taken a look at the implementation and I think the performance penalty is minimal.  I think there's no increase in terms of circuit complexity.  The only thing I see is that a step preemptively calculates all the cells based on `MAX_STEP_HEIGHT`, and there's also a cache that depends on `MAX_STEP_HEIGHT`, so there may be a small increase in memory consumption, but I don't think that's significant.

Reply:
BTW here's an idea on how to reduce the number or cells:
Currently you're allocating 32 cells for `k`, `a_reduced`, `d`, and `e`.  From these words, `d` and `e` are only used in the form of 64 bit limbs.  `k` and `a_reduced` are used in `LowerThan` gadgets so this may not apply.

So `d` and `e` could just be 4 x 64 bit cells, and used directly in the `MulAddWords512Gadget`.  Of course this requires changing the `MulAddWords512Gadget` gadget.

But with this you'd a difference of `- 32 * 2 + 4 * 2 = - 56` cells.

Reply:
I'll work on this on a separate PR to stop blocking the merging of `ADDMOD`. I think this suggestion may also apply to the `ModGadget` so a good bit could be optimized.

Reply:
> I've taken a look at the implementation and I think the performance penalty is minimal. I think there's no increase in terms of circuit complexity. The only thing I see is that a step preemptively calculates all the cells based on `MAX_STEP_HEIGHT`, and there's also a cache that depends on `MAX_STEP_HEIGHT`, so there may be a small increase in memory consumption, but I don't think that's significant.

Actually I realized there's a penalty!  By increasing `MAX_STEP_HEIGHT` we increase the number of rotations!  But as far as I understand, the EVM circuit is already heavy on rotations, and by implementing SHPLONK the high number of rotations will not be an issue.